extensions/common/vault/vault-aws/src/test/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVaultExtensionTest.java

```diff
@@ -47,4 +47,15 @@ void configOptionRegionProvided_shouldNotThrowException() {
         extension.createVault(validContext);
     }
 
+    @Test
+    void configOptionEndpointOverrideProvided_shouldNotThrowException() {
```

Comment:
this test is asserting nothing. I know that you inherited from the already existing tests, but tests without assertions should be avoided.
Demonstration: if you try to add a proper assertion like:

```java
        var vault = extension.createVault(validContext);

        assertThat(vault).extracting("smClient", type(SecretsManagerClient.class)).satisfies(client -> {
            assertThat(client.serviceClientConfiguration().endpointOverride()).contains(URI.create("http://localhost:4566"));
        });
```

it won't pass, that's because you set the mock to respond a call where only the setting key is passed, but not the default value.
Another suggestion would be not to mock the config object but to create one using `ConfigFactory.fromMap`, that will make everything easier.